src/box/field_default_func.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "port.h"
+#include "engine.h"
+#include "tt_static.h"
+#include "tuple_format.h"
+#include "field_default_func.h"
+
+/**
+ * Find `default_func` in func cache.
+ * Return NULL if something went wrong (diag is set).
+ */
+struct func *
+field_default_func_find(struct field_default_func *default_func)
+{
+	struct func *func = func_by_id(default_func->id);
+	if (func == NULL) {
+		diag_set(ClientError, ER_NO_SUCH_FUNCTION,
+			 int2str(default_func->id));
+	}
+	return func;
+}
+
+/**
+ * Check `func` for compliance with the field default function rules.
+ *
+ * Returns 0 on success. On error, sets diag and returns -1.
+ * `fieldno` is used for better error message.
+ */
+static int
+field_default_func_verify(struct func *func, uint32_t fieldno)
+{
+	const char *func_name = func->def->name;
+
+	if (func->def->language == FUNC_LANGUAGE_LUA &&
+	    func->def->body == NULL) {
+		diag_set(ClientError, ER_WRONG_SPACE_FORMAT,
+			 fieldno + TUPLE_INDEX_BASE,
+			 tt_sprintf("field default Lua function '%s' must have "
+				    "persistent body", func_name));
```

Comment:
Nit: Not sure it's worth bothering passing the field no around. Maybe better introduce a new error code for invalid default value function, like we did for constraints? The code would have the following format: `Invalid default value function '%s': %s`.

Reply:
OK, changed to `ER_CREATE_DEFAULT_FUNC`